sklearn/metrics/tests/test_classification.py

```diff
@@ -1384,6 +1383,31 @@ def test_log_loss():
     loss = log_loss(y_true, y_pred)
     assert_almost_equal(loss, 1.0383217, decimal=6)
 
+    # case when len(np.unique(y_true)) != y_pred.shape[1]
+    y_true = [1,2,2]
+    y_pred = [[0.2, 0.7, 0.3], [0.6, 0.5, 0.3], [0.3, 0.9, 0.1]]
+    loss = log_loss(y_true, y_pred, labels=[1,2,3])
+    assert_almost_equal(loss, 1.0630345, decimal=6)
+
+    # test labels option
+
+    y_true = [2, 2]
+    y_pred = [[0.2, 0.7], [0.6, 0.5]]
+    y_score = np.array([[0.1, 0.9], [0.1, 0.9]])
+    error_str = ('y_true has only one label. Please provide '
+                 'the true labels explicitly through the labels argument.')
+    assert_raise_message(ValueError, error_str, log_loss, y_true, y_pred)
+
+    y_pred = [[0.2, 0.7], [0.6, 0.5], [0.2, 0.3]]
```

Comment:
hmm i don't see why this should be renamed to y_score2? could you explain a bit more. sorry if im being slow / missing something obvious